src/sentry/testutils/factories.py

```diff
@@ -617,6 +620,27 @@ def create_useremail(user, email, **kwargs):
 
         return useremail
 
+    @staticmethod
+    def inject_performance_problems(jobs, _):
+        for job in jobs:
+            job["performance_problems"] = []
+            for f in job["data"]["fingerprint"]:
+                f_data = f.split("-")
+                group_type = GroupType(int(f_data[0]))
+                perf_fingerprint = f_data[1]
```

Comment:
the format is `group_type-fingerprint`

Reply:
I might be [misreading the code](https://github.com/getsentry/sentry/blob/master/src/sentry/utils/performance_issues/performance_detection.py#L1002-L1007) but the format looks like it's `1-{group_type}-{actual_fingerprint}` (hard-coded 1)

Reply:
Yeah this is confusing. We originally wanted to parse an outputted fingerprint to be able to test versioning (that hardcoded 1 is supposed to be a fingerprint strategy version) and see what group type it is. The problem is that we can't save unhashed fingerprints. In group creation step, we end up hashing the fingerprint again because we can't store something like this `1-{group_type}-{actual_fingerprint}`.

Here I'm not following that format for fingerprint. For testing purposes, I'm passing a fingerprint that consists of a group type and an actual group fingerprint. I'm doing that to ensure I get performance group with the type I need and a fingerprint that I need.

Reply:
we should update the outputted fingerprint format to avoid further confusion